samples/extensions/extended_dynamic_state2/extended_dynamic_state2.cpp

```diff
@@ -764,59 +752,48 @@ glm::vec4 ExtendedDynamicState2::get_changed_alpha(const vkb::sg::PBRMaterial *o
 }
 
 /**
- * @fn void ExtendedDynamicState2::scene_pipeline_divide(std::vector<std::vector<SceneNode>> *scene_node)
+ * @fn void ExtendedDynamicState2::scene_pipeline_divide(std::vector<SceneNode> const &scene_node)
  * @brief Spliting main scene into two separate.
  * @details This operation is required to use same "draw_from_scene" function to draw models that are using different
  * 			pipelines (baseline and tessellation)
  */
-void ExtendedDynamicState2::scene_pipeline_divide(std::vector<std::vector<SceneNode>> *scene_node)
+void ExtendedDynamicState2::scene_pipeline_divide(std::vector<SceneNode> const &scene_node)
 {
-	int scene_nodes_cnt = scene_node->at(SCENE_ALL_OBJ_INDEX).size();
-
-	std::vector<SceneNode> scene_elements_baseline;
-	std::vector<SceneNode> scene_elements_tess;
-
 	/* Divide main scene to two (baseline and tessellation) */
-	for (int i = 0; i < scene_nodes_cnt; i++)
+	for (int i = 0; i < scene_node.size(); i++)
 	{
-		if (scene_node->at(SCENE_ALL_OBJ_INDEX).at(i).name == "Geosphere")
+		if (scene_node.at(i).name == "Geosphere")
 		{
-			scene_elements_tess.push_back(scene_node->at(SCENE_ALL_OBJ_INDEX).at(i));
+			scene_elements_tess.push_back(scene_node.at(i));
 		}
 		else
 		{
-			scene_elements_baseline.push_back(scene_node->at(SCENE_ALL_OBJ_INDEX).at(i));
+			scene_elements_baseline.push_back(scene_node.at(i));
 		}
 	}
-
-	scene_node->push_back(scene_elements_baseline);
-	scene_node->push_back(scene_elements_tess);
 }
 
 /**
- * @fn void ExtendedDynamicState2::draw_from_scene(VkCommandBuffer command_buffer, std::vector<std::vector<SceneNode>> *scene_node, sceneObjType_t scene_index)
+ * @fn void ExtendedDynamicState2::draw_from_scene(VkCommandBuffer command_buffer, std::vector<SceneNode> const &scene_node)
  * @brief Drawing all objects included to scene that is passed as argument of this function
  */
-void ExtendedDynamicState2::draw_from_scene(VkCommandBuffer command_buffer, std::vector<std::vector<SceneNode>> *scene_node, sceneObjType_t scene_index)
+void ExtendedDynamicState2::draw_from_scene(VkCommandBuffer command_buffer, std::vector<SceneNode> const &scene_node)
 {
-	auto &node               = scene_node->at(scene_index);
-	int   scene_elements_cnt = scene_node->at(scene_index).size();
-
-	for (int i = 0; i < scene_elements_cnt; i++)
+	for (int i = 0; i < scene_node.size(); i++)
 	{
-		const auto &vertex_buffer_pos    = node[i].sub_mesh->vertex_buffers.at("position");
-		const auto &vertex_buffer_normal = node[i].sub_mesh->vertex_buffers.at("normal");
-		auto &      index_buffer         = node[i].sub_mesh->index_buffer;
+		const auto &vertex_buffer_pos    = scene_node[i].sub_mesh->vertex_buffers.at("position");
+		const auto &vertex_buffer_normal = scene_node[i].sub_mesh->vertex_buffers.at("normal");
+		auto       &index_buffer         = scene_node[i].sub_mesh->index_buffer;
 
-		if (scene_index == SCENE_BASELINE_OBJ_INDEX)
+		if (scene_node.at(i).name != "Geosphere")
```

Comment:
Maybe we could use one way of accessing vector scene_node[i] or scene_node.at(i)